packages/algolia/src/index.ts

```diff
@@ -140,103 +136,111 @@ interface IndexToAlgoliaNextraOptions {
+  if (!sitemapUrls?.length) {
+    console.log(sitemapXmlPath, JSON.stringify(sitemap, null, '  '));
+    throw new Error('Sitemap urls not found at path urlset>url');
+  }
+  const urls = sitemapUrls.map(({ loc }) => loc).filter(Boolean);
+  if (!urls?.length) {
+    console.log(sitemapXmlPath, JSON.stringify(sitemap, null, '  '));
+    throw new Error('No sitemap urls found path urlset>url>loc');
   }
+  return urls;
 }
 
 export async function nextraToAlgoliaRecords({
   docsBaseDir,
   source,
   domain,
+  sitemapXmlPath,
   objectsPrefix = new GitHubSlugger().slug(source),
 }: IndexToAlgoliaNextraOptions): Promise<AlgoliaRecord[]> {
+  const sitemapUrls = await getSitemapUrlsXml(sitemapXmlPath);
+
   const objects: AlgoliaRecord[] = [];
   const slugger = new GitHubSlugger();
 
-  // cache for all needed `_meta.json` files
-  const metadataCache: Record<string, any> = {};
-
-  const getMetadataForFile = async (
-    filePath: string,
-  ): Promise<[title: string, hierarchy: string[], urlPath: string] | void> => {
-    const hierarchy = [];
-
-    const fileDir = filePath.split('/').slice(0, -1).join('/');
-    const fileName = filePath.split('/').pop()!;
-    const folders = filePath
-      .replace(docsBaseDir, '')
-      .replace(fileName, '')
-      .split('/')
-      .filter(Boolean);
-    // docs/guides/advanced -> ['Guides', 'Advanced']
-    // by reading meta from:
-    //  - docs/guides/_meta.json (for 'advanced' folder)
-    //  - docs/_meta.json (for 'guides' folder)
-    while (folders.length) {
-      const folderPath = folders.join('/');
-      const folder = folders.pop()!;
-
-      metadataCache[folderPath] ||= await getMetaFromFile(
-        path.join(docsBaseDir, folderPath, '_meta.json'),
-      );
-      const folderName = metadataCache[folderPath][folder];
-      const resolvedFolderName =
-        typeof folderName === 'string' ? folderName : folderName?.title || folder;
-      if (resolvedFolderName) {
-        hierarchy.unshift(resolvedFolderName);
+  for (const sitemapUrl of sitemapUrls) {
+    const pageSlug = withoutTrailingSlashes(sitemapUrl.replace(withTrailingSlash(domain), ''));
+
+    let pagePath = path.join(docsBaseDir, pageSlug);
+    let pageContent: string;
+
+    // since we strip extensions from urls, if the path exists - it's a directory, otherwise a markdown file
+    const isDir = existsSync(pagePath);
+    pagePath = `${pagePath}${isDir ? '/index' : ''}.md`;
+    if (existsSync(pagePath)) {
+      // .md
+      pageContent = await readFile(pagePath, 'utf-8');
+    } else if (((pagePath = `${pagePath}x`), existsSync(pagePath))) {
+      // .mdx
+      pageContent = await readFile(pagePath, 'utf-8');
+    } else {
+      throw new Error(`Page ${pagePath.replace('.mdx', '.{md,mdx}')} does not exist`);
+    }
+
+    // front-matter title or first appearing H1 heading content
+    const {
+      data: { title: matterTitle, type: matterType },
```

Comment:
I don’t think we use frontmatter `type` somewhere in docs

Reply:
Let's leave it just in case